lib/internal/Magento/Framework/Stdlib/DateTime/Timezone.php

```diff
@@ -174,7 +183,14 @@ public function date($date = null, $locale = null, $useTimezone = true, $include
                     $timeType,
                     new \DateTimeZone($timezone)
                 );
-                $date = $formatter->parse($date) ?: (new \DateTime($date))->getTimestamp();
+
+                $localeDate = !empty($locale) ? $locale : null;
+                $date = $formatter->parse($date) ?: (new \Zend_Date(
```

Comment:
Still not solved: https://github.com/magento/magento2/pull/11306#discussion_r143856096

Reply:
I'm trying to solve this problem, but it's not easy and it will take me a while fix it.